src/background.js

```diff
@@ -128,22 +128,23 @@ function unregisterContentScripts () {
 
 async function registerContentScripts () {
   CONTENT_SCRIPTS.forEach((contentScript) => {
+    contentScript.files
+        .filter(file => file.match(/.*\.css$/))
+        .map(file => `${contentScript.folder}/${file}`)
+        .map(file => browser.tabs.insertCSS(null, { file }));
+
+    GLOBAL_SCRIPTS.map(file => browser.tabs.executeScript(null, { file }));
+    contentScript.files
+      .filter(file => file.match(/.*\.js$/))
+      .map(file => `${contentScript.folder}/${file}`)
+      .map(file => browser.tabs.executeScript(null, { file }))

```

Comment:
I see. I'm new to these `browser.* API` methods.
The `browser.contentScripts.register` doesn't seem to work on Chrome, even after the polyfill. I'll do some more testing on to why it's that

Reply:
I looked into this today. `browser.contentScripts.register` isn't supported by Chrome, so the polyfill doesn't implement it. It only polyfills Chrome's existing api and nothing else.

It's possible that the script loading strategy needs to change to support Chrome. This extension will have to filter the content scripts before injecting them with the cross-platform `browser.tabs.executeScript`.

I've been slowly working on a larger refactor of the extension that redoes how it loads content scripts, so I can move all the supported sites to `optionalPermissions`. I think Chrome support should wait for this refactor to finish since it uses `browser.tabs.onActivate` to determine which scripts to inject, which is the exact functionality needed to move away from leaning on `matches` in `browser.contentScripts.register`.

Reply:
OK. makes sense. I tried another polyfill and keeping `browser.contentScripts.register` but didn't work
https://github.com/fregante/content-scripts-register-polyfill